esphome/components/udp/sensor.py

```diff
@@ -0,0 +1,27 @@
+import esphome.codegen as cg
+from esphome.components.sensor import new_sensor, sensor_schema
+from esphome.config_validation import All, has_at_least_one_key
+from esphome.const import CONF_ID
+
+from . import (
+    CONF_PROVIDER,
+    CONF_REMOTE_ID,
+    CONF_UDP_ID,
+    SENSOR_SCHEMA,
+    require_internal_with_name,
+)
+
+AUTO_LOAD = ["udp"]
```

Comment:
A sensor requires a provider to be set in the main component right? In which case this should be a dependency, not auto loaded
```suggestion
DEPENDENCIES = ["udp"]
```